src/coreclr/scripts/superpmi_benchmarks.py

```diff
@@ -192,11 +206,25 @@ def build_and_run(coreclr_args, output_mch_name):
          "--framework", "net8.0", "--no-restore", "/p:NuGetPackageRoot=" + artifacts_packages_directory,
          "-o", artifacts_directory], _exit_on_fail=True)
 
-    # Disable ReadyToRun so we always JIT R2R methods and collect them
-    collection_command = f"{dotnet_exe} {benchmarks_dll}  --filter \"*\" --corerun {os.path.join(core_root, corerun_exe)} --partition-count {partition_count} " \
-                         f"--partition-index {partition_index} --envVars DOTNET_JitName:{shim_name} " \
-                         " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
-                         "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
+    if benchmark_binary.lower().startswith("microbenchmarks"):
+        # Disable ReadyToRun so we always JIT R2R methods and collect them
+        collection_command = f"{dotnet_exe} {benchmarks_dll}  --filter \"*\" --corerun {os.path.join(core_root, corerun_exe)} --partition-count {partition_count} " \
+                            f"--partition-index {partition_index} --envVars DOTNET_JitName:{shim_name} " \
+                            " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
+                            "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
+    elif benchmark_binary.lower().startswith("demobenchmarks"):
+        # Disable ReadyToRun so we always JIT R2R methods and collect them
+        collection_command = f"{dotnet_exe} {benchmarks_dll}  -f *CollisionBatcherTaskBenchmarks.* *GroupedCollisionTesterBenchmarks.* *GatherScatterBenchmarks.* " \
+                            " *OneBodyConstraintBenchmarks.* *TwoBodyConstraintBenchmarks.* *ThreeBodyConstraintBenchmarks.* *FourBodyConstraintBenchmarks.* " \
+                            " *SweepBenchmarks.* *ShapeRayBenchmarks.* *ShapePileBenchmark.* *RagdollTubeBenchmark.* " \
+                            f" --corerun {os.path.join(core_root, corerun_exe)} --envVars DOTNET_JitName:{shim_name} " \
+                            " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
+                            "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
```

Comment:
How long would it take to run these normally (no BDN strategy alterations)?

Reply:
Thanks @LoopedBard3 for getting these numbers:

```
    ML.NET ~ 12 min
    FSharp ~8 min
    Bepuphysics ~6.5 min
    ImageSharp ~6 min
    Illink ~16.5 min
    Roslyn ~ 17 mins.
```

Reply:
So 56 mins.

In the job you link to above the entire thing took 49 minutes on windows, so the cost above doesn't seem too bad?

<img width="911" alt="image" src="https://github.com/dotnet/runtime/assets/10121823/7f3bb7a9-13d7-4296-a7b2-95ed4e801191">

Reply:
The 49 mins includes the queueing time. The actual execution with `invocationCount 1` is couple of minutes.

Reply:
`-IterationCount 1`:

<img width="1351" alt="image" src="https://github.com/dotnet/runtime/assets/12488060/b3bde76f-51e2-4622-b5aa-0f1d97beb0ba">


default args, so definitely the collection part slows it down further. I will merge this PR as-is.

<img width="1443" alt="image" src="https://github.com/dotnet/runtime/assets/12488060/1b9a35d2-22d5-498d-8b7e-19e2be895ec7">